src/Control/Monad/Trace.hs

```diff
@@ -186,59 +193,66 @@ instance MonadTransControl TraceT where
   liftWith = defaultLiftWith TraceT traceTReader
   restoreT = defaultRestoreT TraceT
 
-instance (MonadBaseControl IO m, MonadIO m) => MonadTrace (TraceT m) where
+traceWith :: Builder -> Scope -> (Scope -> IO a) -> IO a
+traceWith bldr parentScope f = do
+  let mbParentSpn = scopeSpan parentScope
```

Comment:
Aren't all these `liftIO` calls redundant?

Reply:
Of course they are :man_facepalming: Fixed, as well as the `liftBase`s in `addSpanEntryWith`.